src/app/containers/CpsAssetPageMain/index.test.jsx

```diff
@@ -1,24 +1,37 @@
 import React from 'react';
-import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import { StaticRouter } from 'react-router-dom';
+import matchSnapshot from './testHelpers';
 import { ServiceContextProvider } from '#contexts/ServiceContext';
 import { RequestContextProvider } from '#contexts/RequestContext';
 import CpsAssetPageMain from '.';
+import preprocessor from '#lib/utilities/preprocessor';
+import convertToOptimoBlocks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks';
 import pidginPageData from '#data/pidgin/cpsAssets/tori-49450859';
 
+const pageData = preprocessor(pidginPageData, [convertToOptimoBlocks]);
+
 describe('CpsAssetPageMain', () => {
-  shouldMatchSnapshot(
-    'should match snapshot',
-    <ServiceContextProvider service="pidgin">
-      <RequestContextProvider
-        bbcOrigin="https://www.test.bbc.co.uk"
-        isAmp={false}
-        pageType="MAP"
-        pathname="/pidgin/tori-49450859"
-        service="pidgin"
-        statusCode={200}
-      >
-        <CpsAssetPageMain service="pidgin" pageData={pidginPageData} />
-      </RequestContextProvider>
-    </ServiceContextProvider>,
-  );
+  it('should match snapshot', async done => {
+    const component = (
+      /*
+        for the value it would bring, it is much simpler to wrap a react-router Link in a Router, rather than mock a Router or pass some mocked context.
+      */
+      <StaticRouter>
+        <ServiceContextProvider service="pidgin">
+          <RequestContextProvider
+            bbcOrigin="https://www.test.bbc.co.uk"
+            isAmp={false}
+            pageType="MAP"
+            pathname="/pidgin/tori-49450859"
+            service="pidgin"
+            statusCode={200}
+          >
+            <CpsAssetPageMain service="pidgin" pageData={await pageData} />
+          </RequestContextProvider>
+        </ServiceContextProvider>
+      </StaticRouter>
+    );
+
+    matchSnapshot(component, done);
+  });
 });
```

Comment:
Could you not have just done

```jsx
import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import CpsAssetPageMain from '.';
import preprocessor from '#lib/utilities/preprocessor';
import convertToOptimoBlocks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks';
import pidginPageData from '#data/pidgin/cpsAssets/tori-49450859';

const pageData = preprocessor(pidginPageData, [convertToOptimoBlocks]);

describe('CpsAssetPageMain', () => {
  shouldMatchSnapshot(
    'should match snapshot',
    <StaticRouter>
      <ServiceContextProvider service="pidgin">
        <RequestContextProvider
          bbcOrigin="https://www.test.bbc.co.uk"
          isAmp={false}
          pageType="MAP"
          pathname="/pidgin/tori-49450859"
          service="pidgin"
          statusCode={200}
        >
          <CpsAssetPageMain service="pidgin" pageData={pageData} />
        </RequestContextProvider>
      </ServiceContextProvider>
    </StaticRouter>,
  );
});

```

Reply:
`preprocessor(pidginPageData, [convertToOptimoBlocks])` is now aysnc, so `pageData` is a promise which needs to resolve before it is passed to `<CpsAssetPageMain>` as out containers shouldn't be handling promises. This all stems from the fact we preprocessor the fixture data before our snapshot tests.